### ISSUE TITLE
GPU not detected (using CPU instead)

### ISSUE THREAD

**Issue by uddee** [uddee/GPU-Accelerated-AI-Model-Optimization#1]
Description:
I am running the code provided in the repository, but it is not utilizing the GPU for model training. Instead, the code is running on the CPU, even though my system has a CUDA-compatible GPU. The expected behavior is that the code should detect the GPU and run the model on it.

Expected Behavior:
The code should detect the GPU if available and run the model on the GPU. The output should include GPU memory usage and training on the GPU.

Actual Behavior:
The code is running on the CPU instead of the GPU, and there is no GPU memory usage being displayed.

Training loss: 2.311065912246704
Training time: 0.0563 seconds
Using device: cpu

Steps to Reproduce:
Clone the repository and install dependencies.
Run the provided code (gpu_optimization_example.ipynb).
Observe that the model runs on the CPU instead of the GPU.

Environment:
OS: [e.g., Windows 10, Ubuntu 20.04]
Python version: [e.g., 3.8.5]
PyTorch version: [e.g., 1.9.0]
CUDA version: [e.g., 11.2]
GPU model: [e.g., NVIDIA GeForce GTX 1080]
Driver version: [e.g., 465.19.01]
